connectors/es/client.py

```diff
@@ -125,7 +125,8 @@ async def check_exists(self, indices=None, pipelines=None):
             except NotFoundError:
                 raise PreflightCheckError(f"Could not find pipeline {pipeline}")
 
-    async def delete_indices(self, indices=None):
-        if indices is None:
-            indices = []
+    async def delete_indices(self, indices):
+        if indices is None or len(indices) == 0:
```

Comment:
You are still allowing `indices` to be `None` -- do we really want to support a call to `delete_indices` to be a no-op ?

Reply:
Yes, the caller can still call `an_index.delete_indices(indices=None)`

Reply:
> > where you check that the indices are not empty, so I think `delete_indices` can directly pass the call to `self.client`
> 
> Indeed, but I don't think we should define the method based on it's caller's implementation.

But we're not implementing an API for hypothetical ways of calling it -- it's added for actual use cases, and right now the one single use case is to delete indices. Why would you call it if you don't intend to delete some indices? what would be the actual use case of passing `None` to it?  I don't see that requirement/need, and I think allowing `None` and silently ignoring it by returning is a recipe for problems.